tlsfuzzer/extract.py

```diff
@@ -168,26 +174,56 @@ def _parse_pcap(self):
         with open(self.capture, 'rb') as pcap:
             capture = dpkt.pcap.Reader(pcap)
 
+            syn_ack_seq = 0
+
+            # since timestamp is Decimal() we don't have to worry about float()
+            # precision
             for timestamp, pkt in capture:
                 link_packet = dpkt.ethernet.Ethernet(pkt)
                 ip_pkt = link_packet.data
                 tcp_pkt = ip_pkt.data
 
+                if (tcp_pkt.flags & dpkt.tcp.TH_SYN and
+                        tcp_pkt.dport == self.port and
+                        ip_pkt.dst == self.ip_address):
```

Comment:
on one hand, yes, on the other hand, in actuality it should be rewritten further so that it can handle rearranged and duplicated packets, so I don't want to invest too much time into it.

Also, half of the indentation is because of open context manager, exception handling and a simple for loop to process the packets in order, so it's not actually very complex.

So I'd rather leave it as-is.